yacc/reader.c

```diff
@@ -915,7 +764,7 @@ declare_tokens(int assoc)
         if (isalpha(c) || c == '_' || c == '.' || c == '$')
             bp = get_name();
         else if (c == '\'' || c == '"')
-            bp = get_literal();
+            invalid_literal(lineno, line, cptr);
```

Comment:
Actually I'm not sure I follow what is going on here, can you explain?
I don't understand the details of `get_literal`, but it seems to accept quoted string literals, and you are turning this into an error message here, why?

Reply:
IIUC this feature comes from C, and generated code like

```c
enum Token {
   A = 'x',
};
```

This worked because in C, both enums and character constants are really just integers.  In OCaml this would generate something like

```ocaml
type v = A of 'x'
       | B of "y"
```

which is an OCaml syntax error.